statsmodels/discrete/tests/test_predict.py

```diff
@@ -378,6 +378,32 @@ def test_distr(case):
 
     assert_allclose(res.resid_pearson, (y2 - mean) / np.sqrt(var_), rtol=1e-13)
 
+   
+    # which dict for distributions
+    
+    which_dict = {
+        Logit: ['mean', 'var', 'prob', 'linear'],
+        Probit: ['mean', 'var', 'prob', 'linear'],
+        ZeroInflatedPoisson: ['mean', 'var', 'prob', 'linear'],
+        ZeroInflatedGeneralizedPoisson: ['mean', 'var', 'prob', 'linear'],
+        ZeroInflatedNegativeBinomialP: ['mean', 'var', 'prob', 'linear'],
+        GeneralizedPoisson: ['mean', 'var', 'prob', 'linear'],
+        NegativeBinomial: ['mean', 'var', 'prob', 'linear']
+    }
+
+    # loop through and test
+    
+    if cls_model in which_dict:
+        # for each which, conduct smoke and shape test
+        for which in which_dict[cls_model]:
+            probs = res.predict(which=which, y_values=np.arange(5))
```

Comment:
you need to condition here on `if which == "prob"`
only "prob" uses `y_values` and returns 2-dim, others return 1-dim with length `nobs = len(mod.endog)`